spell/spell-reviewer-goerli-checklist.md

```diff
@@ -135,37 +132,53 @@ Spell Actions:
     * [ ] Increase Ilk Debt Ceiling (set DC + increase Global DC)
     * [ ] `bump` `RwaLiquidationOracle` with new computed increased price (`val`)
-      * [ ] `val` should enable DAI to be drawn over the loan period while taking into
-            account the configured `ink` amount, interest rate and liquidation ratio
+      * [ ] `val` should enable DAI to be drawn over the loan period while taking into account the configured `ink` amount, interest rate and liquidation ratio (see below)
+        * [ ] New `val` is calculated with `line * [(1 + duty) ** years] * mat` - rounded up - and makes sense in context of the [rate mechanism](https://github.com/makerdao/developerguides/blob/master/mcd/intro-rate-mechanism/intro-rate-mechanism.md). Minimum duration is usually in the exec doc of the spell with the RWAXXX ilk onboarding.
+        * [ ] Comment explaining `val` formula (`Debt ceiling * [ (1 + RWA stability fee ) ^ (minimum deal duration in years) ] * liquidation ratio`) is present
+        * [ ] Accompanying comment above `bump` line in format `// XXM * 1.XX^X * X.XX as a WAD` corresponding to the `val` calculation formula (e.g. `// 15M * 1.03^2 * 1.00 as a WAD`) is present along with the calculation formula on the line above
+        * [ ] IF combining `val` of multiple RWA ilks being combined, `val` calculation is done once per ilk and added to make the total, with workings provided in code comments. The existing `val` value can be retrieved by calling `read()` on `PIP_RWAXX` and converting the result into decimal.
     * [ ] Poke `spotter` to pull in the new price
-* [ ] Payments
-  * [ ] Streams (`DssVest`)
-    * [ ] `DssVest` Interface is correct
-    * [ ] Ensure that `cap` > max new vest `tot`/`tau` otherwise file cap as well
-    * [ ] Timestampts match Doc (`bgn`, `fin`)
-    * [ ] CUs Addresses match Doc (`usr`)
-    * [ ] Amount matches Doc (`tot`, if decimals are present consider using `ether`)
-    * [ ] Vesting Duration matches Doc (`tau`)
-    * [ ] Cliff Duration matches Doc (`eta`)
-    * [ ] Restricted (by default)
-    * [ ] Manager match Doc (`mgr`, set to zero for DAI streams by default)
-    * [ ] IF [DssVestTransferrable](https://github.com/makerdao/dss-vest/blob/master/src/DssVest.sol#L463) is used
-      * [ ] Ensure `DssVestTransferrable` allowance is increased by new vesting delta (by approving the `transferrable` vest contract to allowance + new total amount streamed)
-  * [ ] CUs MKR Transfers
-    * [ ] Recipient Addresses match Doc
-    * [ ] Transfers Amounts match Doc
-    * [ ] Follows Previous Patterns
-  * [ ] Direct SB DAI Payment
-    * [ ] Recipient Addresses match Doc
-    * [ ] Payment Amounts match Doc
-    * [ ] Follow Previous Patterns
-  * [ ] Ensure Recipient Addresses match `addresses_wallets.sol`
+* [ ] SubDAO Content
+  * [ ] SubDAO SubProxy spell execution
+    * [ ] SubDAO spell address matches exec doc
+    * [ ] SubDAO spell deployer is in `addresses_deployers`
```

Comment:
I don't think we should have this requirement.

Should we care about who deploys a subdao spell?

I'm currently deep diving into the spell test suite, and the only time the addresses in `addresses_deployers.sol` are used is in the [`checkAuth` test](https://github.com/makerdao/spells-mainnet/blob/c1f3caa1374e4f911db6a246852947c2b76692b3/src/DssSpell.t.base.sol#L1529-L1545).

This test basically fetches all contracts in the chainlog and checks if all deployers have been `deny`ed in those.

Spells don't make it into the chainlog, so having the deployer of a spell there does not do anything other than maybe keeping a list there. This might be important for MakerDAO spells, for the sake of the spell-crafting process, but I don't think we should have SubDAO spells deployers there.

Reply:
cc @SidestreamColdMelon @The-Arbiter @SidestreamSweatyPumpkin @0xdecr1pto

Reply:
If you're OK with at least documenting it somewhere I am OK with that given what you've mentioned here (and our discussions last week on this item).

> Should we care about who deploys a subdao spell?

Only if they can affect the core system / contracts we care about - which happens to be what this test checks.

That being said, it _does_ (parabolically at best) increase test execution time, so I think it isn't the end of the world to not check that Auth for these EOAs.

_However:_
If the address is not an EOA (it is a multisig, a factory, etc.) then I think we should check this.

I also think we should have a log of the deployer addresses so that we don't get attacked by a similar looking deployer address doing something, so there should at the very least be a log of which deployer addresses were used in the past.

If you look at the addresses_deployers.sol on Goerli, it is much more labelled than the one on Mainnet because people in the past decided that it wasn't necessary to label addresses for the mainnet branch, so that's the impetus behind my thoughts on this (I don't want to make that 'mistake' again).

Either way, the SubDAO actions were put together quickly and so this does seem like an extraneous (or at best, minimally helpful) check, so I will modify it a bit:

```
* [ ] SubDAO spell deployer is a smart contract (e.g. multisig or factory)
      * [ ] Ensure the deployer address is in `addresses_deployers.sol` as an entry
    * [ ] SubDAO spell deployer is purely an EOA
      * [ ] Ensure the deployer address is in `addresses_deployers.sol` as a comment
```

This solves the problems:

- No additional test running time
- Deployer address is still _transparently_ available (and accessible by third parties who wish to audit spells for whatever reason)
- Information regarding contract deployers is in one place (the addresses_deployers.sol file)

I still think that auth checks here are necessary for the following reason:

When the MakerDAO-level spell is calling 'execute' on something (not a DelegateCall), this can only be done comfortably because it is known that the spell contract is not authed on any core contracts. 
The only way to auth something on a core contract is to already be `relied` on said core contract.
One of the only ways that a spell being 'executed' can be authed on a core contract is if some address (such as an EOA) auths the spell, allowing it to perform potentially harmful actions.

Of course, the EOA could do it themself, which is why EOAs are (to my knowledge) never authed on core contracts within MCD, _however_ it is foreseeable that Multisigs for SubDAOs may be treated as more 'trusted' actors (which are not strictly EOAs if they're smart contracts) and therefore might have auth on something core (even if it's peripheral rather than something vital like Vat), so these checks might prevent something in those cases (obviously they aren't currently happening though).

Either way, this is reworked for now so that it does not impact test execution time while preserving transparency and documenting (potential) hazards.